tcplistener.go

```diff
@@ -148,6 +147,18 @@ func handleListenedConn(address string, conn *net.TCPConn, headerByteSize int, m
 	headerBuffer := make([]byte, headerByteSize)
 	dataBuffer := make([]byte, maxMessageSize)
 	for {
+		// Before we try reading a message, check to see if we've been shutdown or not
+		// That way, shutting down will allow any current message to finish, but block
+		// any receives on the connection after it occurs, in addition to blocking new
+		// connections from being accepted, above in the initial connection code.
+		select {
+		case <-sdChan:
+			conn.Close()
+			return
+		default:
+			// Nothing, continue on to accept incoming bytes
+		}
+
```

Comment:
I think an `atomic.LoadInt64` flag check might be a good fit for cases where there isn't another channel we're racing with. But I'm not sure you want that here, I think hitting this condition might be an exception rather than the norm, usually we'll fly right by and sit in the blocking `readFromConnection`.

I believe this `TCPConn` should be forcibly closed outside of this function in response to the overall shutdown channel being closed.